buildscripts/autogen_sources_methods.py

```diff
@@ -0,0 +1,182 @@
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *****************************************************************************
+'''
+
+docstring_header = '''\
+"""
+
+| This file contains overloads for various extension types auto-generated with autogen_sources_methods.py
+
+"""
+'''
+
+arithmetic_binops_symbols = {
+    'add': '+',
+    'div': '/',
+    'sub': '-',
+    'mul': '*',
+    'truediv': '/',
+    'floordiv': '//',
+    'mod': '%',
+    'pow': '**',
+}
+
+comparison_binops_symbols = {
+    'lt': '<',
+    'gt': '>',
+    'le': '<=',
+    'ge': '>=',
+    'ne': '!=',
+    'eq': '==',
+}
+
+target_rel_filename = 'sdc/sdc_autogenerated.py'
+
+
+if __name__ == '__main__':
+
+    sdc_root_path = Path(__file__).absolute().parents[1]
+    target_file_path = sdc_root_path.joinpath(target_rel_filename)
+
+    # read templates_module as text and extract import section to be copied into auto-generated target file
+    module_text = inspect.getsource(templates_module)
+    module_text_lines = module_text.splitlines(keepends=True)
+
+    # extract copyright text from templates file
+    copyright_line_numbers = [k for k, line in enumerate(module_text_lines) if '# *****' in line]
+    copyright_section_text = ''.join(module_text_lines[copyright_line_numbers[0]: copyright_line_numbers[1] + 1])
+
+    # extract copyright text from templates file - this only works if imports in it
+    # are placed contigiously, i.e. at one place and not intermixed with code
+    imports_line_numbers = [k for k, line in enumerate(module_text_lines) if 'import ' in line]
+    imports_start_line, import_end_line = min(imports_line_numbers), max(imports_line_numbers)
+    import_section_text = ''.join(module_text_lines[imports_start_line: import_end_line + 1])
+
+    series_operator_comp_binop = templates_module.sdc_pandas_series_operator_comp_binop
+    # read function templates for arithmetic and comparison operators from templates module
+    template_series_arithmetic_binop = inspect.getsource(templates_module.sdc_pandas_series_binop)
+    template_series_comparison_binop = inspect.getsource(templates_module.sdc_pandas_series_comp_binop)
+    template_series_arithmetic_binop_operator = inspect.getsource(templates_module.sdc_pandas_series_operator_binop)
+    template_series_comparison_binop_operator = series_operator_comp_binop
```

Comment:
Shouldn't the template_series_comparison_binop_operator be assigned inspect.getsource(...)?
I tried to run autogen_source_methos.py but got the following:
```
(SDC_MASTER) C:\Users\akozlov\AppData\Local\Continuum\anaconda3\hpat>python buil
dscripts\autogen_sources_methods.py
c:\users\akozlov\appdata\local\continuum\anaconda3\hpat\sdc\utilities\utils.py:4
59: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if init_vals is not ():
Exception of type AttributeError: 'function' object has no attribute 'replace'
while writing to a file: C:\Users\akozlov\AppData\Local\Continuum\anaconda3\hpat
\sdc\sdc_autogenerated.py
```